### ISSUE TITLE
Docker Deployment Script

### ISSUE THREAD

**Issue by Morgul** [RocketChat/Rocket.Chat#770]
Right now, to deploy as a docker with `docker-compose`, you have to either checkout the repo, or pull down the `docker-compose.yml`. Additionally, the docker-compose file is configured to bind to port 80, but if you're running an nginx proxy in front, you probably want it to bind to some other port. Finally, it's also nice to set your own restart strategy (ex: `restart=always`).

What if, instead, we took a page from Docker's book, and hosted a shell script on `http://get.rocket.chat`?

## The Crazy Idea

```bash
$ curl -sSL http://get.rocket.chat/ | sh
```

This would have the following prerequisites:

* `curl` is installed. (It is in most ubuntu server installs these days)
* `docker` is installed. (Obviously.)
* `docker-compose` is installed. (Not required or included by docker, but is a pretty simple script to get.)

Those honestly aren't that hefty, requirements wise. 

The impressive is what this would do under the hood. We'd pull the `docker-compose.yml` from the git repo (master or from a tagged version), modify the script to set any options, and then stream it into `docker-compose up -d`. We can then support options like this:

```bash
$ curl -sSL http://get.rocket.chat/ | PORT=443 RESTART='always' sh
```

It's still just one line to get up and running, but now we've configured everything for the user.

### Issues

One question this raises is, how does a user upgrade? We could save the built docker compose file somewhere, like `/opt/rocket.chat`, or something. We'd need to document it, but then they could update versions with `docker-compose up -d`, which is pretty nice.

**Comment by geekgonecrazy** [comment]
This is a pretty cool approach.

Ping @Sing-Li

**Comment by Sing-Li** [comment]
Please take a look at  the rocket.chat image. Not the dockerfile on github source dir , and not rocket chat/rocket.chat image.

The rocket.chat image .....That is the official way we are looking to support docker. Docker liked it, so do we. I know the image is old.... we are working hard to get the automated build pipeline in place.

We definitely do not want to do any meteor build or checkout the source....that's for development, and ridiculous for deployment.

I think HA configuration and strategy should be external to the deployment/distribution strategy as it usually involves very different building blocks .... Typically independent of the underlying distribution technology (ie docker).

**Comment by biomassives** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  acmeideal@gmail.com
    Domain biomassiv.es has exceeded the max emails per hour (100/100 (100%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext3.iad.github.net ([192.30.252.194]:49258 helo=github-smtp2b-ext-cp1-prd.iad.github.net)
	by server23.websitehostserver.net with esmtps (TLSv1.2:DHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.85)
	(envelope-from <noreply@github.com>)
	id 1ZaY39-0040D7-NE
	for greg@biomassiv.es; Fri, 11 Sep 2015 18:48:12 -0500
Date: Fri, 11 Sep 2015 16:48:14 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1442015294;
	bh=burqG1AGjLnEmMYRoOJiUc1yxBQGEKna7RaE1+j5weY=;
	h=From:Reply-To:To:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=0FLeqKqxBYbmQqHjYN8JgZob/Pv7VVNcf4RTnmFgsLUsubTZ3eadIcnqNXEuDOlhx
	 U5xhBLEsKqsNHGOjI5ABnzUDX7352zr9qk8TOMq93GHKpHOZtxSoKcdrSbQjrzQAMP
	 0OQ5EEfkIdHouwgOiWQV1VK001r1MxjXuJ55e+vA=

**Comment by geekgonecrazy** [comment]
Pruned the message returned from invalid email.

Docker-compose aside.. Being able to install with such a simple method is what I find potentially the most useful.  Granted this does require a level of trust that not everyone will have.  But those could install via another method.

**Comment by Sing-Li** [comment]
Agreed.   The `curl xxxx |   sh`  method is great with or without docker.   And certainly can be adapted for either development or even some deployment cases.

**Comment by Morgul** [comment]
@Sing-Li, @geekgonecrazy: That's basically what I was trying to drive at. I was originally focused on docker (probably because I just spend the last 40+ hours setting up a half dozen automatic deployment scenarios using docker at work), but you're right, it doesn't have to be docker.

Regardless of the deployment strategy, we need to make sure that whatever the deployment strategy is, there's a straightforward, documented method for upgrades, but that's manageable, I suspect. :)

**Comment by Sing-Li** [comment]
Thanks, @Morgul. :+1:  Sorry I forgot to acknowledge the great idea :smile: (personally still slightly upset about the lifeblood server being down)

@gureilha is our documentation specialist and he is currently tidying up and organizing ALL the different deployment docs we have everywhere.   It does reflect on the diverse user base (installers, hosters, consultants, developers, experimenters, evaluators and so on) we have.   Thank you for your understanding.

**Comment by Sing-Li** [comment]
Yay!  Itz back :smile:

**Comment by biomassives** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  acmeideal@gmail.com
    Domain biomassiv.es has exceeded the max emails per hour (117/100 (117%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext2.iad.github.net ([192.30.252.193]:48645 helo=github-smtp2b-ext-cp1-prd.iad.github.net)
	by server23.websitehostserver.net with esmtps (TLSv1.2:DHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.85)
	(envelope-from <noreply@github.com>)
	id 1ZaYpq-002OCt-T4
	for greg@biomassiv.es; Fri, 11 Sep 2015 19:38:32 -0500
Date: Fri, 11 Sep 2015 17:38:33 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1442018314;
	bh=XLzar6Skd5SxOa3Km0n+N3yiaE0M1vYEhwWGOYdpwl4=;
	h=From:Reply-To:To:Cc:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=w+oDTQbxmdP0/gpkzfl0S4iZRY8G/2C8Pty95umqWOD2LmpvVG8z/g7L1RbNg2pkN
	 F47+Ft9gph8oWY9ZN7mnDLyrcTAfjOTrQQ1KnM4F4K9neeT7iqaOLhDbNwcxUqf6Ko
	 nZW+C/5E1gPbU0hY8C6DYaqFYCDDNYLAcEPKTIBY=

**Comment by engelgabriel** [comment]
@geekgonecrazy  I thinks this has been done, right?

**Comment by geekgonecrazy** [comment]
As far as documenting upgrade paths I think most if not all have been.  

But as to the original issue.. I still think this would be a really great way to do.  Even if it only did tarball install and not docker.  

I'll move to feature requests with original topic idea
